manifest-build-tools/application/release_to_atlas.py

```diff
@@ -0,0 +1,242 @@
+        """
+        create_provider_url = self.generate_url("create_provider", atlas_version)
+        provider_data = {'provider[name]': provider}
+        resp = self.session.post(create_provider_url, data=provider_data)
+        if resp.ok:
+            print "Create box provider {0} of version {1} successfully.".format(provider, atlas_version)
+        else:
+            print "Failed to create box provider.\n {0}".format(resp.text)
+            sys.exit(1)
+
+    def upload_box(self, atlas_version, provider, box_file):
+        """
+        Upload one box to a specific version/provider
+        """
+        upload_box_url = self.generate_url("upload_box", atlas_version, provider)
+        resp = self.session.get(upload_box_url)
+        if resp.ok:
+            upload_path = resp.json()["upload_path"]
+            cmd = "curl -X PUT --upload-file " +  box_file  + " " +  upload_path
+            p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+            retval = p.wait()
+            if retval == 0 :
+                print "Upload box {0} to version/{1}/provider/{2} successfully!".format(box_file, atlas_version, provider)
+            else:
+                print "Failed to Upload box {0} to version/{1}/provider/{2}!\n {3}".format(box_file, atlas_version, provider,retval)
+                sys.exit(1)
+
+    def version_exist(self, atlas_version):
+        """
+        Check if box version exists
+        """
+        check_version_url = self.generate_url("check_version", atlas_version)
+        resp = self.session.get(check_version_url)
+        if resp.ok:
+            print "Box version {0} already exists.".format(atlas_version)
+            return True
+        print "Box version {0} doesn't' exist, will be created soon.".format(atlas_version)
+        return False
+
+    def provider_exist(self, atlas_version, provider):
+        """
+        Check if box provider exists.
+        NOTICE: provider depends on a specific box version.
+        """
+        if not self.version_exist(atlas_version):
+            print "Box version {0} doesn't' exist, please create version before check provider.".format(atlas_version)
+            return False
+        check_provider_url = self.generate_url("check_provider", atlas_version, provider)
+        resp = self.session.get(check_provider_url)
+        if resp.ok:
+            print "{0} provider of version {1} already exists!".format(provider, atlas_version)
+            return True
+        print "{0} provider of version {1} doesn't' exist, will be created soon".format(provider, atlas_version)
+        return False
+
+    def generate_url(self, purpose, atlas_version=None, provider=None):
+        """
+        Tool method, Generate all possible urls according to purpose
+        """
+        purpose_handler = {
+            "check_version": lambda atlas_version, provider: "/".join([self.atlas_url, self.box, "version/{0}".format(atlas_version)]),
```

Comment:
First I think use a ```dict``` is more clear than ```swith case```, but at last I find this style(anonymous function in python) is complex too....
So be it.